src/test/prevector_tests.cpp

```diff
@@ -188,61 +187,59 @@ BOOST_AUTO_TEST_CASE(PrevectorTestInt)
     for (int j = 0; j < 64; j++) {
         prevector_tester<8, int> test;
         for (int i = 0; i < 2048; i++) {
-            int r = insecure_rand();
-            if ((r % 4) == 0) {
-                test.insert(insecure_rand() % (test.size() + 1), insecure_rand());
+            if (InsecureRandBits(2) == 0) {
+                test.insert(InsecureRandRange(test.size() + 1), InsecureRand32());
             }
-            if (test.size() > 0 && ((r >> 2) % 4) == 1) {
-                test.erase(insecure_rand() % test.size());
+            if (test.size() > 0 && InsecureRandBits(2) == 1) {
+                test.erase(InsecureRandRange(test.size()));
             }
-            if (((r >> 4) % 8) == 2) {
-                int new_size = std::max<int>(0, std::min<int>(30, test.size() + (insecure_rand() % 5) - 2));
+            if (InsecureRandBits(3) == 2) {
```

Comment:
This is calling `InsecureRandBits(3)` multiple times rather than using the same `r`, which is not equivalent to the original code. (Previously the cases were mutually exclusive.)

Reply:
Indeed; this matches the change made upstream to these tests (which are from upstream).

Reply:
Hmm, ok. The resulting code is weird because it still compared the randomness to distinct values in each branch, which is why I thought the upstream change might be incorrect. But looking more closely I guess it doesn't actually matter, because the effect is only to potentially make more calls to methods of `test`.

Reply:
IIRC this was noted in the upstream PR (the distinct values could all just be 0, as their only purpose now is to sample a probability) but was deemed too noisy to fix.